plugins/modules/ec2_vpc_nacl.py

```diff
@@ -255,42 +253,102 @@ def tags_changed(nacl_id, client, module):
     )
 
 
-def rules_changed(aws_rules, param_rules, Egress, nacl_id, client, module):
+def ansible_to_boto3_dict_rule(ansible_rule: List[Any], egress: bool) -> Dict[str, Any]:
+    boto3_rule = {}
+    if isinstance(ansible_rule, list):
+        boto3_rule["RuleNumber"] = ansible_rule[0]
+        boto3_rule["Protocol"] = str(PROTOCOL_NUMBERS[ansible_rule[1]])
+        boto3_rule["RuleAction"] = ansible_rule[2]
+        boto3_rule["Egress"] = egress
+        if is_ipv6(ansible_rule[3]):
+            boto3_rule["Ipv6CidrBlock"] = ansible_rule[3]
+        else:
+            boto3_rule["CidrBlock"] = ansible_rule[3]
+        if icmp_present(ansible_rule):
+            boto3_rule["IcmpTypeCode"] = {"Type": int(ansible_rule[4]), "Code": int(ansible_rule[5])}
+        else:
+            if ansible_rule[6] or ansible_rule[7]:
+                boto3_rule["PortRange"] = {"From": ansible_rule[6], "To": ansible_rule[7]}
+    return boto3_rule
+
+
+def diff_network_acl_rules(
```

Comment:
What about splitting this function into one for finding all rules to add and the other for finding all rules to remove?

Reply:
done